base/task.jl

```diff
@@ -5,7 +5,56 @@
 # allow tasks to be constructed with arbitrary function objects
 Task(f) = Task(()->f())
 
-show(io::IO, t::Task) = print(io, "Task ($(t.state)) @0x$(hex(convert(UInt, pointer_from_objref(t)), WORD_SIZE>>2))")
+function show(io::IO, t::Task)
+    print(io, "Task ($(t.state)) @0x$(hex(convert(UInt, pointer_from_objref(t)), WORD_SIZE>>2))")
+    if t.state == :failed
+        show(io, CapturedException(t.result, t.backtrace))
+    end
+end
+
+type CapturedException
```

Comment:
A `CapturedException` is a container type for an exception and its backtrace. I don't expect it to be thrown by itself and hence did not make it a subtype of `Exception`. Actually, since we can throw any object, I am not really clear what decides if an object should be a subtype of `Exception`.

`CompositeException` however should be a subtype of `Exception`. Will change it.